doc/api/environment-map.mkd

```diff
@@ -0,0 +1,58 @@
+# R10K::API Environment Maps
+
+An environment map (or envmap) is a data structure which represents the desired or actual state of a single Puppet environment. It is designed to be a language-neutral data structure as much as possible, so things like Ruby's insertion-order key sorting should not be relied upon for any meaningful state. Envmaps are designed to be used in a functional manner, so any operation which decorates or mutates an envmap should return a new copy instead of making in-place modifications. Envmaps should also serialize cleanly to and from JSON.
+
+Full example of an "abstract" (unresolved) envmap:
+
+```json
+{
+  "environment": "production",
+  "source": <Hash representation of R10K::Source>,
```

Comment:
Yeah, my intention was basically just to capture enough information to be able to deploy the right control repo snapshot, so it should probably be an explicitly pared down representation of the source. (Which I think is roughly what I had in my first iteration of this... :))